lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -293,14 +295,16 @@ impl ShardReplicaSet {
                     let Some(snapshot_manifest) = snapshot_manifest.get(segment_id) else {
                         log::debug!("Removing outdated segment {}", segment_path.display());
 
-                        tokio_fs::remove_dir_all(&segment_path)
-                            .await
-                            .map_err(|err| {
-                                CollectionError::service_error(format!(
-                                    "failed to remove outdated segment {}: {err}",
-                                    segment_path.display(),
-                                ))
-                            })?;
+                        tokio::task::spawn_blocking({
```

Comment:
would it be possible to avoid `spawn_blocking`? We had those problems with not enough threads recently, so if we can do it async, let's do it async

Reply:
WDYM?

`tokio::fs::delete_dir_all` is just `std::fs::delete_dir_all` inside `spawn_blocking`. (just like other `tokio::fs` wrappers)
In my code I did the same, just more verbose on the caller side.